File: ECE-FIAD2 (Backend B)/Dominio/Entidades/Evoluciones/Evolucion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Dominio.Entidades.Bases;
using Dominio.Entidades.Doctores;
using Dominio.Entidades.HistoriasClinicas;
namespace Dominio.Entidades.Evoluciones
{
    public class Evolucion : EntidadBase
    {
        public int IdHistoriaClinica { get; set; }
        public int IdDoctor { get; set; }
        public DateTime Fecha { get; set; } = DateTime.Now;
        public string Diagnostico { get; set; } = string.Empty;
        public string Tratamiento { get; set; } = string.Empty;
        public string Notas { get; set; } = string.Empty;
        public virtual HistoriaClinica HistoriaClinica { get; set; } = null!;
        public virtual Doctor Doctor { get; set; } = null!;
    }
}

File: GestionClinica-Frontend-A/src/types/Evolucion/Evolucion.ts
import { type Doctor } from "../Doctor/Doctor";
import { type HistoriaClinica } from "../HistoriaClinica/HistoriaClinica";

export interface Evolucion {
    id: number;
    idHistoriaClinica: number;
    idDoctor: number;
    fecha: string;
    diagnostico: string;
    tratamiento: string;
    notas?: string;
    activo: boolean;
    doctor?: Doctor;
    historiaClinica?: HistoriaClinica;
}